app/models/vacation_request.rb

```diff
@@ -36,4 +36,39 @@ class VacationRequest < ActiveRecord::Base
     :unpaid,
     :sickness
   ]
+
+  def cannot_intersect_with_other_vacations
+    number_of_records  = number_of_intersected_records
+    number_of_records += number_of_records_that_include_vacation
+
+    errors.add(:base, 'cannot inresect with other vacations')\
+      if number_of_records > 0
+  end
+
+private
+
+  # Number of records that somehow intersect with the vacation
+  # For example, the vacation '2015-09-01'..'2015-09-10'
+  # intersects with the following vacations:
+  #   - '2015-09-05'..'2015-09-15', by '2015-09-05'
+  #   - '2015-08-25'..'2015-09-10', by '2015-09-10'
+  #   - '2015-09-05'..'2015-09-09', by both
+  # The last vacation is actually included by the vacation in subject.
+  def number_of_intersected_records
+    VacationRequest.where(
+      VacationRequest.where(start_date: start_date..actual_end_date)
+        .where(actual_end_date: start_date..actual_end_date)
+        .where_values
+        .reduce(:or)
+    ).where(user_id: user_id).count
+  end
+
+  # Number of records that includes the vacation
+  # For example, the vacation '2015-09-01'..'2015-09-20'
+  # includes the subject vacation '2015-09-05'..'2015-09-15'
+  def number_of_records_that_include_vacation
```

Comment:
The following solution is OK at the moment:
```
VacationRequest
  .where('start_date BETWEEN ? AND ? OR actual_end_date BETWEEN ? AND ? OR start_date <= ? AND actual_end_date >= ?', start_date, actual_end_date, start_date, actual_end_date, start_date, actual_end_date)
  .where(user_id: user_id).count
```
But there must be more flexible solution, with Arel, https://github.com/rails/arel.